File: src/main/java/io/awilson/timein/services/SessionServiceImpl.java
package io.awilson.timein.services;

import io.awilson.timein.domain.Session;
import io.awilson.timein.repositories.SessionRepository;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class SessionServiceImpl implements SessionService {

    @Setter(onMethod = @_(@Autowired))
    private SessionRepository sessionRepository;

    @Override
    public Iterable<Session> listAllSessions() {
        return sessionRepository.findAll();
    }

    @Override
    public Session getSessionById(Integer id) {
        return sessionRepository.findOne(id);
    }

    @Override
    public Session saveSession(Session session) {
        return sessionRepository.save(session);
    }

    @Override
    public void deleteSession(Integer id) {
        sessionRepository.delete(id);
    }
}

File: src/main/java/io/awilson/timein/services/SessionServiceImpl.kt
package io.awilson.timein.services

import io.awilson.timein.domain.Session
import io.awilson.timein.repositories.SessionRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class SessionServiceImpl : SessionService {

    @Autowired
    lateinit var repo: SessionRepository

    override fun listAllSessions(): Iterable<Session> = repo.findAll()

    override fun getSessionById(id: Int): Session = repo.findOne(id) ?: throw Exception("Session:$id not found")

    override fun saveSession(session: Session): Session = repo.save(session)

    override fun deleteSession(id: Int) = repo.delete(id)
}